### PR TITLE
DOC: Instructions for installing with dependencies from `conda`

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR adds instructions for building oneDAL from source using packages from `conda` to satisfy dependencies instead of using system installs. Packages from `conda` don't come with activation scripts, so this adds the necessary instructions for getting them to work with the oneDAL make file.

_Disclaimer: I haven't fully tested the windows instructions._

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.

**Performance**

Not applicable.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -35,12 +35,17 @@ Note: the Intel(R) oneAPI components listed here can be installed together throu
 
 https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit.html
 
+These dependencies can also be installed through the `conda` software, but doing so will require a few additional setup steps - see rest of the instructions for details.
```
Comment (Vika-F) [inline INSTALL.md:38]: I think it is possible to add the link to a respective chapter and remove "see rest ...".

File: INSTALL.md [hunk 2]
```diff
@@ -35,12 +35,17 @@ Note: the Intel(R) oneAPI components listed here can be installed together throu
...
+
 ## Docker Development Environment
 
 [Docker file](https://github.com/uxlfoundation/oneDAL/tree/main/dev/docker) with the oneDAL development environment
 is available as an alternative to the manual setup.
 
 ## Installation Steps
+
+_(If using `conda`, see subsequent sections for instructions)_
```
Comment (Vika-F) [inline INSTALL.md:47]: I don't think this is needed if the reference to the chapter is added at the beginning.

File: INSTALL.md [hunk 3]
```diff
@@ -195,3 +200,69 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
 ```shell
 ./_cmake_results/intel_intel64_so/adaboost_dense_batch
 ```
+
+## Installation with a conda environment
```
Comment (Vika-F) [inline INSTALL.md:204]: Let's rename to something like: `Conda Development Environment Setup`, and remove the last part about running `make` that duplicates the info in the previous chapter.

File: INSTALL.md [hunk 4]
```diff
@@ -195,3 +200,69 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+    -c https://software.repos.intel.com/python/conda/ \ `# Intel's repository`
+    -c conda-forge \ `# conda-forge, for tools like 'make'`
+    make python \ `# used by the build system`
+    dpcpp-cpp-rt dpcpp_linux-64 intel-sycl-rt \ `# Intel compiler packages`
+    tbb tbb-devel \ `# required TBB packages`
+    mkl mkl-devel mkl-static mkl-dpcpp mkl-devel-dpcpp \ `# required MKL packages`
+    cmake `# required to build the examples only`
+```
+
+_(for windows, replace `dpcpp_linux-64` with `dpcpp_win-64`, and remove the comments within backticks)_
```
Comment (Vika-F) [inline INSTALL.md:228]: Let's have separate instructions for Linux and Windows - save the time for users on copy-pasting.

Also, i think the minimal supported versions for the packages (at least some) need to be added. Like python=3.9, etc.

File: INSTALL.md [hunk 5]
```diff
@@ -195,3 +200,69 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+
+Then, one needs to modify environment variables as appropriate to point to the necessary libraries - **assuming a linux system:**
```
Comment (Vika-F) [inline INSTALL.md:230]: Other chapters use '(Linux*)' and '(Windows*)' to distinguish. Let's align and use the wording like this:

```suggestion
Modify the environment variables to point to the necessary libraries (Linux *):
```

File: INSTALL.md [hunk 6]
```diff
@@ -195,3 +200,69 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+export TBBROOT=${CONDA_PREFIX}
+export LD_LIBRARY_PATH="${CONDA_PREFIX}/lib:${LD_LIBRARY_PATH}"
+export LIBRARY_PATH="${CONDA_PREFIX}/lib:${LIBRARY_PATH}"
+export CPATH="${CONDA_PREFIX}/include:${CPATH}"
+export PATH="${CONDA_PREFIX}/bin:${PATH}"
+export PKG_CONFIG_PATH="${CONDA_PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH}"
+export CMAKE_PREFIX_PATH="${CONDA_PREFIX}/lib/cmake:${CMAKE_PREFIX_PATH}"
+```
+
+Equivalent for windows:
```
Comment (Vika-F) [inline INSTALL.md:243]:
```suggestion
Modify the environment variables to point to the necessary libraries (Windows*):
```

File: INSTALL.md (hunk 1 above)
Comment (david-cortes-intel) [reply INSTALL.md]: Added a link.

File: INSTALL.md (hunk 2 above)
Comment (david-cortes-intel) [reply INSTALL.md]: Removed.

File: INSTALL.md (hunk 3 above)
Comment (david-cortes-intel) [reply INSTALL.md]: Renamed.

File: INSTALL.md (hunk 4 above)
Comment (david-cortes-intel) [reply INSTALL.md]: Changed to separate code snippets.

File: INSTALL.md (hunk 5 above)
Comment (david-cortes-intel) [reply INSTALL.md]: Splitted into linux/windows sections.

File: INSTALL.md [hunk 7]
```diff
@@ -195,3 +199,72 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+First, create a conda environment for building oneDAL, after `conda` has been installed:
+
+```shell
+conda create -y -n onedal_env
+conda activate onedal_env
+```
+
+Then, install the necessary dependencies from the appropriate channels with `conda`:
+
+* Linux:
```
Comment (Vika-F) [inline INSTALL.md:216]:
```suggestion
**Linux\***:
```

File: INSTALL.md [hunk 8]
```diff
@@ -195,3 +199,72 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+    -c https://software.repos.intel.com/python/conda/ \ `# Intel's repository`
+    -c conda-forge \ `# conda-forge, for tools like 'make'`
+    make python>=3.9 \ `# used by the build system`
+    dpcpp-cpp-rt dpcpp_linux-64 intel-sycl-rt \ `# Intel compiler packages`
+    tbb tbb-devel \ `# required TBB packages`
+    mkl mkl-devel mkl-static mkl-dpcpp mkl-devel-dpcpp \ `# required MKL packages`
+    cmake `# required to build the examples only`
+```
+
+* Windows:
```
Comment (Vika-F) [inline INSTALL.md:229]:
```suggestion
**Windows\***:
```

File: INSTALL.md [hunk 9]
```diff
@@ -195,3 +199,72 @@ For example, in a Linux platform, assuming one wishes to execute the `adaboost_d
...
+    make dos2unix python>=3.9^
+    dpcpp-cpp-rt dpcpp_win-64 intel-sycl-rt^
+    tbb tbb-devel^
+    mkl mkl-devel mkl-static mkl-dpcpp mkl-devel-dpcpp^
+    cmake
+```
+
+Then modify the relevant environment variables to point to the conda-installed libraries:
+
+* Linux:
```
Comment (Vika-F) [inline INSTALL.md:244]:
```suggestion
**Linux\***:
```

**Review by Vika-F — COMMENTED** [review]
The final request and lgtm.

File: INSTALL.md (hunk 7 above)
Comment (david-cortes-intel) [reply INSTALL.md:216]: That doesn't appear to render correctly:
image

Comment (david-cortes-intel) [reply INSTALL.md]: Modified the 4 occurrences to use the formatting you're suggesting but inside bullet points.

**Review by Vika-F — APPROVED** [review]

**Pull request merged**